### PR TITLE
refactor: [M3-6352] - MUI v5 Migration - `Components > LabelAndTagsPanel`

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Refactored `LabelAndTagsPanel` by removing jss, converting to functional component.

## Major Changes 🔄
- Converted from class to functional component
- Removed render guards
- Converted tests from Enzyme to RTL and added some more tests

## Preview 📷
Screenshot 2023-06-16 at 10 52 38 AM

## How to test 🧪
1. Ensure everything renders and functionality is unchanged
2. Run tests `yarn test packages/manager/src/components/LabelAndTagsPanel/LabelAndTagsPanel.test.tsx`

### CODE REVIEW

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/components/LabelAndTagsPanel/LabelAndTagsPanel.tsx [hunk 1]
```diff
@@ -1,60 +1,40 @@
...
-    },
-    expPanelButton: {
-      padding: 0,
-      marginTop: theme.spacing(2),
-    },
-  });
+import { useTheme } from '@mui/material/styles';
 
-const styled = withStyles(styles);
 interface Props {
```
Comment (mjac0bs) [inline packages/manager/src/components/LabelAndTagsPanel/LabelAndTagsPanel.tsx:8]:
```suggestion
interface LabelAndTagsProps {
```

File: packages/manager/src/components/LabelAndTagsPanel/LabelAndTagsPanel.tsx [hunk 2]
```diff
@@ -1,60 +1,40 @@
...
-          })}
-          data-qa-label-input
-          noMarginTop
-        />
-        {tagsInputProps && <TagsInput {...tagsInputProps} />}
-      </Paper>
-    );
-  }
-}
+export const LabelAndTagsPanel = (props: Props) => {
```
Comment (mjac0bs) [inline packages/manager/src/components/LabelAndTagsPanel/LabelAndTagsPanel.tsx:14]:
```suggestion
export const LabelAndTagsPanel = (props: LabelAndTagsProps) => {
```

**Review by mjac0bs — APPROVED** [review]
Functionality and test looked good!

**Comment by jaalah-akamai** [comment]
Screenshot 2023-06-20 at 1 18 13 PM

Cypress test looks good

**Pull request merged**